tool/tctl/common/plugin/entraid.go

```diff
@@ -0,0 +1,389 @@
+	entraapiutils "github.com/gravitational/teleport/api/utils/entraid"
+	"github.com/gravitational/teleport/lib/integrations/azureoidc"
+	"github.com/gravitational/teleport/lib/utils/oidc"
+	"github.com/gravitational/teleport/lib/web/scripts/oneoff"
+)
+
+type entraArgs struct {
+	cmd                  *kingpin.CmdClause
+	authConnectorName    string
+	defaultOwners        []string
+	useSystemCredentials bool
+	accessGraph          bool
+	force                bool
+}
+
+func (p *PluginsCommand) initInstallEntra(parent *kingpin.CmdClause) {
+	p.install.entraID.cmd = parent.Command("entraid", "Install an EntraId integration.")
+	cmd := p.install.entraID.cmd
+	cmd.
+		Flag("name", "Name of the plugin resource to create").
+		Default("entra-id").
+		StringVar(&p.install.name)
+
+	cmd.
+		Flag("auth-connector-name", "Name of the SAML connector resource to create").
+		Default("entra-id-default").
+		StringVar(&p.install.entraID.authConnectorName)
+
+	cmd.
+		Flag("use-system-credentials", "Uses system credentials instead of OIDC.").
+		BoolVar(&p.install.entraID.useSystemCredentials)
+
+	cmd.Flag("default-owner", "List of Teleport users that are default owners for the imported access lists. Multiple flags allowed.").
+		Required().
+		StringsVar(&p.install.entraID.defaultOwners)
+
+	cmd.
+		Flag("access-graph", "Enables Access Graph cache build.").
+		Default("true").
+		BoolVar(&p.install.entraID.accessGraph)
+
+	cmd.
+		Flag("force", "Proceed with installation even if plugin already exists.").
+		Short('f').
+		Default("false").
+		BoolVar(&p.install.entraID.force)
+}
+
+type entraSettings struct {
+	accessGraphCache *azureoidc.TAGInfoCache
+	clientID         string
+	tenantID         string
+}
+
+var (
+	errCancel = trace.BadParameter("operation canceled")
+)
+
+func (p *PluginsCommand) entraSetupGuide(proxyPublicAddr string) (entraSettings, error) {
+	pwd, err := os.Getwd()
+	if err != nil {
+		return entraSettings{}, trace.Wrap(err)
+	}
+	f, err := os.CreateTemp(pwd, "entraid-setup-*.sh")
+	if err != nil {
+		return entraSettings{}, trace.Wrap(err, "failed to create temp file")
+	}
+
+	defer os.Remove(f.Name())
+
+	buildScript, err := buildScript(proxyPublicAddr, p.install.entraID.authConnectorName, p.install.entraID.accessGraph, p.install.entraID.useSystemCredentials)
+	if err != nil {
+		return entraSettings{}, trace.Wrap(err, "failed to build script")
+	}
+
+	if _, err := f.Write([]byte(buildScript)); err != nil {
+		return entraSettings{}, trace.Wrap(err, "failed to write script to file")
+	}
+
+	if err := f.Close(); err != nil {
+		return entraSettings{}, trace.Wrap(err, "failed to close file")
+	}
+	fileLoc := f.Name()
+
+	bold := color.New(color.Bold).SprintFunc()
+	boldRed := color.New(color.Bold, color.FgRed).SprintFunc()
+
+	tmpl := `Step 1: Run the Setup Script
+
+1. Open ` + bold("Azure Cloud Shell") + ` (Bash) using ` + bold("Google Chrome") + ` or ` + bold("Safari") + ` for the best compatibility.
```

Comment:
If you click on this link, it should land you in the Shell
```
https://portal.azure.com/#cloudshell/
```

I'm saying this because if you go to Azure console, you won't see any reference to Azure Cloud Shell unless you already know what it is and click on the little terminal icon on the top right.

<img width="517" alt="image" src="https://github.com/user-attachments/assets/35ffbf95-b265-4cab-a917-7545b05a92bb">